### PR TITLE
feat: implement std.minArray

### PR DESCRIPTION
**Pull request by DeepanshuGoel17**
Add std.minArray function in standard library.

PR for go-jsonnet: https://github.com/google/go-jsonnet/pull/685

### CODE REVIEW

**Comment by DeepanshuGoel17** [comment]
@sparkprime Can you please take a look at this?

File: stdlib/std.jsonnet [hunk 1]
```diff
@@ -1703,6 +1703,16 @@ limitations under the License.
 
   sum(arr):: std.foldl(function(a, b) a + b, arr, 0),
 
+  minArray(arr)::
```
Comment (CertainLach) [inline stdlib/std.jsonnet:1706]: Wouldn't it be more useful with keyF argument, as in std.sort/std.set/std.uniq?

Comment (DeepanshuGoel17) [reply stdlib/std.jsonnet]: Correct, I have made the required changes. Thanks!

**Comment by DeepanshuGoel17** [comment]
@CertainLach Can you please review the changes?

File: stdlib/std.jsonnet [hunk 2]
```diff
@@ -1703,6 +1703,16 @@ limitations under the License.
 
   sum(arr):: std.foldl(function(a, b) a + b, arr, 0),
 
+  minArray(arr, keyF=id)::
+    assert std.length(arr) > 0 : 'Expected atleast one element in array. Got none';
```
Comment (CertainLach) [inline stdlib/std.jsonnet:1707]: For slightly better ergonomics...
```
minArray(arr, keyF=id, on_empty=throw "expected ..."):: if std.length(arr) == 0 then on_empty else ...
```

File: doc/_stdlib_gen/stdlib-content.jsonnet [hunk 3]
```diff
@@ -1421,6 +1421,16 @@ local html = import 'html.libsonnet';
             |||,
           ]),
         },
+        {
+          name: 'minArray',
+          params: ['arr'],
```
Comment (CertainLach) [inline doc/_stdlib_gen/stdlib-content.jsonnet:1426]: Missing keyF in docs

**Review by CertainLach — CHANGES_REQUESTED** [review]
I'm not this project maintainer... But sure)

File: doc/_stdlib_gen/stdlib-content.jsonnet [hunk 4]
```diff
@@ -1421,6 +1421,16 @@ local html = import 'html.libsonnet';
             |||,
           ]),
         },
+        {
+          name: 'minArray',
+          params: ['arr', 'keyF', 'onEmpty'],
+          availableSince: 'upcoming',
+          description: html.paragraphs([
+            |||
+              Return min of all element in <code>arr</code>.
```
Comment (sparkprime) [inline doc/_stdlib_gen/stdlib-content.jsonnet:1430]: the min of all elements

File: stdlib/std.jsonnet [hunk 5]
```diff
@@ -1703,6 +1703,18 @@ limitations under the License.
 
   sum(arr):: std.foldl(function(a, b) a + b, arr, 0),
 
+  minArray(arr, keyF=id, onEmpty=error 'Expected atleast one element in array. Got none')::
```
Comment (sparkprime) [inline stdlib/std.jsonnet:1706]: at least

Comment (DeepanshuGoel17) [reply stdlib/std.jsonnet]: Done

File: doc/_stdlib_gen/stdlib-content.jsonnet (hunk 4 above)
Comment (DeepanshuGoel17) [reply doc/_stdlib_gen/stdlib-content.jsonnet]: Done

**Pull request merged**